crates/provider/src/chain.rs

```diff
@@ -31,10 +31,16 @@ impl<T: Transport + Clone, N: Network> ChainStreamPoller<T, N> {
     }
 
     pub(crate) fn new(client: WeakClient<T>) -> Self {
+        Self::with_next_yield(client, NO_BLOCK_NUMBER)
+    }
+
+    /// Can be used to force the poller to start at a specific block number.
+    /// Mostly useful for tests.
+    fn with_next_yield(client: WeakClient<T>, next_yield: BlockNumber) -> Self {
         Self {
             client: client.clone(),
             poll_task: PollerBuilder::new(client, "eth_blockNumber", ()),
-            next_yield: NO_BLOCK_NUMBER,
+            next_yield,
             known_blocks: LruCache::new(BLOCK_CACHE_SIZE),
```

Comment:
Also, I'm not quite sure why an LRU cache is used here. Wouldn't a simple `VecDeque` suffice?

Reply:
yeah, imo vecdeque would be more appropriate here actually